components/analytics/true_up_review.tsx

```diff
@@ -0,0 +1,217 @@
+
+import useCanSelfHostedSignup from 'components/common/hooks/useCanSelfHostedSignup';
+import CheckMarkSvg from 'components/widgets/icons/check_mark_icon';
+
+import './true_up_review.scss';
+import {GlobalState} from '@mattermost/types/store';
+import WarningIcon from 'components/widgets/icons/fa_warning_icon';
+import {isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+
+const TrueUpReview: React.FC = () => {
+    const dispatch = useDispatch();
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const isAirGapped = !useCanSelfHostedSignup();
+    const reviewProfile = useSelector(trueUpReviewProfileSelector);
+    const reviewStatus = useSelector(trueUpReviewStatusSelector);
+    const isSystemAdmin = useSelector(isCurrentUserSystemAdmin);
+    const license = useSelector(getLicense);
+    const [submitted, setSubmitted] = useState(false);
+    const trueUpReviewError = useSelector((state: GlobalState) => {
+        const errors = getSelfHostedErrors(state);
+        return Boolean(errors.trueUpReview);
+    });
+
+    useEffect(() => {
+        dispatch(getTrueUpReviewStatus());
+    }, [dispatch, getTrueUpReviewStatus]);
+
+    // Download the review profile as a base64 encoded json file when the review request is submitted.
+    useEffect(() => {
+        if (submitted && isAirGapped) {
+            // Create the bundle as a blob containing base64 encoded json data and assign it to a link element.
+            const blob = new Blob([reviewProfile.content], {type: 'application/text'});
+            const href = URL.createObjectURL(blob);
+
+            const link = document.createElement('a');
+            const date = moment().format('MM-DD-YYYY');
+            link.href = href;
+            link.download = `True Up-${license.Id}-${date}.txt`;
+            document.body.appendChild(link);
+            link.click();
+
+            // Remove link and revoke object url to avoid memory leaks.
+            document.body.removeChild(link);
+            URL.revokeObjectURL(href);
+
+            dispatch(getTrueUpReviewStatus());
+        }
+    }, [isAirGapped, submitted, reviewProfile]);
+
+    const formattedDueDate = (): string => {
+        if (!reviewStatus?.due_date) {
+            return '';
+        }
+
+        // Convert from milliseconds
+        const date = new Date(reviewStatus.due_date);
+        return moment(date).format('MMMM DD, YYYY');
+    };
+
+    const handleSubmitReview = () => {
+        dispatch(getTrueUpReviewBundle());
+        dispatch(getTrueUpReviewStatus());
+        setSubmitted(true);
+    };
+
+    const handleDownloadBundle = () => {
+        dispatch(getTrueUpReviewBundle());
+
+        // Set submitted to true download the bundle data via useEffect
+        // (the selector for the review profile would only return the initial state without doing this).
+        setSubmitted(true);
```

Comment:
This `setSubmitted` should only be called if the `getTrueUpReviewBundle` request is successful. In the actual scenario, the state would be incorrect if the download fails, and this would be to try downloading a file that doesn't exist in the useEffect.